go/test/endtoend/transaction/twopc/fuzzer_test.go

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transaction
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+
+	"vitess.io/vitess/go/mysql"
+)
+
+var (
+	// updateRowsVals are the rows that we use to ensure 1 update on each shard with the same increment.
+	updateRowsVals = [][]int{
+		{
+			4, // 4 maps to 0x20 and ends up in the first shard (-40)
+			6, // 6 maps to 0x60 and ends up in the second shard (40-80)
+			9, // 9 maps to 0x90 and ends up in the third shard (80-)
+			// We can increment all of these values by multiples of 16 and they'll always be in the same shard.
+		},
+	}
```

Comment:
can we avoid this as global, it is modified by the tests and will become unusable for any newer test.

Reply:
yeah, thats a good idea. I'll keep the udpate row vals as a variable in the fuzzer.

Reply:
Fixed